src/Nest/Aggregations/Metric/TopHits/TopHitsAggregate.cs

```diff
@@ -7,38 +8,37 @@ namespace Nest
 {
 	public class TopHitsAggregate : MetricAggregateBase
 	{
-		private readonly IEnumerable<JObject> _hits;
+		[JsonProperty("hits")]
+		private TopHitsProxy _hits { get; set; }
 
-		private readonly JsonSerializer _defaultSerializer;
+		internal JsonSerializer Serializer { get; set; }
 
-		public long Total { get; set; }
-
-		public double? MaxScore { get; set; }
-
-		public TopHitsAggregate() { }
-		
-		internal TopHitsAggregate(IEnumerable<JObject> hits)
-		{
-			_hits = hits;
-		}
-
-		internal TopHitsAggregate(IEnumerable<JObject> hits, JsonSerializer serializer)
-		{
-			_hits = hits;
-			_defaultSerializer = serializer;
-		}
+		public long Total { get { return _hits.Total; } }
+		public double? MaxScore { get { return _hits.MaxScore; } }
 
-		public IEnumerable<Hit<T>> Hits<T>(JsonSerializer serializer = null) 
+		public IEnumerable<Hit<T>> Hits<T>(JsonSerializer serializer = null)
 			where T : class
 		{
-			var s = serializer ?? _defaultSerializer;
-
-			return s != null 
-				? _hits.Select(h => h.ToObject<Hit<T>>(s)) 
-				: _hits.Select(h => h.ToObject<Hit<T>>());
+			var s = serializer ?? this.Serializer;
+			return s != null
+				? _hits.Hits.Select(h => h.ToObject<Hit<T>>(s))
+				: _hits.Hits.Select(h => h.ToObject<Hit<T>>());
 		}
 
 		public IEnumerable<T> Documents<T>(JsonSerializer serializer = null) where T : class =>
 			this.Hits<T>(serializer).Select(h => h.Source);
 	}
+
+	[JsonObject(MemberSerialization.OptIn)]
+	internal class TopHitsProxy
+	{
+		[JsonProperty("total")]
+		public long Total { get; set; }
+
+		[JsonProperty("max_score")]
+		public double? MaxScore { get; set; }
+
+		[JsonProperty("hits")]
+		public IEnumerable<JObject> Hits { get; set; }
```

Comment:
top hits is strongly typed to `T`; if the results could be covariant, should we expose (similar to search) and handle this at deserialization, and still return `Hits<T>` where `T` is the request type `T`?